examples/platform/nrfconnect/util/ThreadUtil.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ThreadUtil.h"
+
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/internal/DeviceNetworkInfo.h>
+
+#include <zephyr.h>
+
+#include <cstring>
+
+void StartDefaultThreadNetwork(void)
+{
+    chip::DeviceLayer::Internal::DeviceNetworkInfo deviceNetworkInfo;
+    memset(&deviceNetworkInfo, 0, sizeof(deviceNetworkInfo));
+
+    const uint8_t masterKey[chip::DeviceLayer::Internal::kThreadNetworkKeyLength] = {0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,\
```

Comment:
This is the default key. I think we can omit it?

Reply:
Unfortunately I think that we can't, because SetThreadProvision() method is not checking the content of ThreadNetworkKey field. If we will not set it, method will copy garbage as a master key (more precisely 0s I guess, as structure is cleared before use).